src/pigpio.cc

```diff
@@ -687,6 +707,205 @@ NAN_METHOD(gpioNotifyClose) {
   }
 }
 
+/* ------------------------------------------------------------------------ */
+/* Waves                                                                    */
+/* ------------------------------------------------------------------------ */
+
+NAN_METHOD(gpioWaveClear) {
+  int rc = gpioWaveClear();
+  if (rc < 0) {
+    return ThrowPigpioError(rc, "gpioWaveClear");
+  }
+  info.GetReturnValue().Set(rc);
+}
+
+NAN_METHOD(gpioWaveAddNew) {
+  int rc = gpioWaveAddNew();
+  if (rc < 0) {
+    return ThrowPigpioError(rc, "gpioWaveAddNew");
+  }
+  info.GetReturnValue().Set(rc);
+}
+
+NAN_METHOD(gpioWaveAddGeneric) {
+  if (info.Length() < 2 || !info[0]->IsUint32() || info[1]->IsNull()) {
+    return Nan::ThrowError(Nan::ErrnoException(EINVAL, "gpioWaveAddGeneric", ""));
+  }
+
+  unsigned numPulses = info[0]->Uint32Value();
+  v8::Local < v8::Array > array = v8::Local < v8::Array > ::Cast(info[1]);
+
+  gpioPulse_t pulses[12000];
+
+  //auto pulses = std::make_unique<gpioPulse_t[]>(array->Length());
+
+  for (unsigned i = 0; i < array->Length(); i++) {
+    if (Nan::Has(array, i).FromJust()) {
+      v8::Local < v8::Object > pulse = v8::Local < v8::Object > ::Cast(array->Get(i)); //Nan::Get(array, i).ToLocalChecked();
+
+      v8::Isolate * isolate = v8::Isolate::GetCurrent();
+
+      v8::Local < v8::Value > _on = pulse->Get(v8::String::NewFromUtf8(isolate, "gpioOn"));
+      v8::Local < v8::Value > _off = pulse->Get(v8::String::NewFromUtf8(isolate, "gpioOff"));
+      v8::Local < v8::Value > _delay = pulse->Get(v8::String::NewFromUtf8(isolate, "usDelay"));
```

Comment:
Verify that _on, _off and _delay are unsigned values by calling IsUint32 for each of them.

Reply:
Fixed with 81a7814